### DISCUSSION TITLE
Battery Mode Returns to default mode - even with Predbat control.

### DISCUSSION THREAD

**Discussion by leacho73** [WillCodeForCats/solaredge-modbus-multi#705] [General]
### Describe the bug

Within the last month, I am getting random issues where my battery mode will automatically restore to its default mode after 60 minutes of the last operation - I am using PredBat to control my charge/discharge - even with setting the timeout value to 10 hours or more - this get's reset after a number of hours and the battery returns to its default mode. It never did this before the 3rd November. Not sure if this is a Solaredge or Predbat issue currently.

### Expected behavior

Predbat should be able to control the battery and it should not return to its default mode, as it didn't before 06:30 on the 3rd November.

### Screenshots

![image](https://github.com/user-attachments/assets/faaa746c-b5a0-40ca-85af-4ab00bf047c7)
![image](https://github.com/user-attachments/assets/69f02c4e-330e-4c44-9efa-70b7ebae1f0a)

### Diagnostic File

-

### Debug logs

```shell
-
```

### Home Assistant Version

2024.11.4

### solaredge-modbus-multi Version

3.0.2

### Installation Type

HAOS

### Read the Instructions

- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] I have read the Known Issues https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues
- [X] This issue is not about a template error (do not report template errors, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by Ashpork** [comment]
I've had/got this issue. Not something this integration would do from my understanding

I found I had two options to resolve this:

- Have predbat change the default storage mode (select.solaredge_i1_storage_default_mode) 
- Setup an automation that spotted the timeout value changed back to the default 1 hour, and reset it to my default of 3 hours.

I often travel, so I didn't go for the first option. I'd rather the system fell back to a default state in the event HA failed and I wasn't available to reset/restore.

**Comment by Noodleyman** [comment]
Morning,

I came here about to ask the same kind of question. I think this relates to the Solar Edge monitor platform now being more aggressive in changing the mode to what it thinks the system should be using.

Mine is now doing it every 10 minutes, and it's becoming a problem, even with automations to revert the previous value. 

I think the SE monitoring platform is every 10 minutes telling the inverter the mode it should be in. This overwrites the mode that it may be using. For example, charge the battery. 

It became a bigger issue for me last night as the battery didn't get charged properly because it kept reverting the charging battery into max self consumption mode. The only options at the monitoring platform level are:
![image](https://github.com/user-attachments/assets/9a6d8bf2-2a9f-4106-b343-1d8bad5c3c8f)

None of those will stop the monitoring platform from sending updates.  (Disabled turns off the battery).  

The only viable option I can seem to find is to block the internet connection for the inverter. but this will block the use of the App as well for a remote view into what the system is doing.

**Comment by Ashpork** [comment]
Are you part of their Grid Flexibility Service?

**Reply by Noodleyman** [reply]
I am not part of the grid flex service as I didn't want them to have control.

**Reply by Ashpork** [reply]
Just thinking of other ideas as to why they might be trying to take control.

**Reply by Noodleyman** [reply]
I like the thinking, but I think it has been this way since I got my systems a few years ago. They used to send the command to restore whatever service they think the monitoring platform would be using about once an hour. That in itself wasn't a massive issue since the automations I had in place caught and handled it as needed.  

I am only guessing, but based on the patterns I am seeing on my system they have just started to spam the command to change the mode a lot more frequently. in some cases every 2 minutes, in some cases every 5 minutes. So whatever I manually set my syste to their cloud system is overwriting. 

I'm going to block it's internet access shortly for a day and validate if that really is what is happening. I'm reasonably confident it is

**Reply by leacho73** [reply]
> I like the thinking, but I think it has been this way since I got my systems a few years ago. They used to send the command to restore whatever service they think the monitoring platform would be using about once an hour. That in itself wasn't a massive issue since the automations I had in place caught and handled it as needed.  
> 
> 
> 
> I am only guessing, but based on the patterns I am seeing on my system they have just started to spam the command to change the mode a lot more frequently. in some cases every 2 minutes, in some cases every 5 minutes. So whatever I manually set my syste to their cloud system is overwriting. 
> 
> 
> 
> I'm going to block it's internet access shortly for a day and validate if that really is what is happening. I'm reasonably confident it is

I'll be interested to see if this fixes the problem - as it's driving me nuts - However - with mine it's convenient that it also happens to be in line with the storage timeout value as to when it reverts to the default mode.

**Reply by jrviz** [reply]
@Noodleyman did you form any conclusions from your tests? I and it seems some other are suffering from this.  I'll try both the default storage mode and automation approaches. I see the concern about HA failing but I have not had any issue with that. I really should not have said that out loud :-)

**Comment by cyberkryten** [comment]
This was caused by a Predbat change and there is a fix for this on Predbat Main, see https://github.com/springfall2008/batpred/issues/1750#issuecomment-2576391426

The change was made as part of some changes to reduce the number of writes to the inverter (due to flash wear caused by register writes on some other inverter types).  

Is anyone aware of whether wear is an issue with writing to either Command Storage Mode or Command Default Mode registers on the SolarEdge inverters?
